### PR TITLE
Adding the basics of material UI and authentication

### PR DESCRIPTION
**Pull request by ataddesse**
This PR:
- Minimizes Angular app build time by modifying the "ng build" command and changing the output folder from frontend/dist to backend/webapp,
- Adds the necessary backend codes for user authentication, 
- Adds Angular Material UI kit via "ng add @angular/material", and
- Adds navbar with sidenave including basic routing needed.

### CODE REVIEW

File: backend/src/main/java/com/google/step/servlets/AuthenticationServlet.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        userService = UserServiceFactory.getUserService();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String url;
+        boolean loggedIn = userService.isUserLoggedIn();
+        if(loggedIn){
+            url = userService.createLogoutURL("/");
+        }
```
Comment (anttsou-google) [inline backend/src/main/java/com/google/step/servlets/AuthenticationServlet.java:31]: nit: put the else with the previous "}" and put a space between. (i.e. } else { ).

File: backend/src/main/java/com/google/step/servlets/WebhookServlet.java [hunk 2]
```diff
@@ -53,14 +56,15 @@
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("application/json;");
+    userAuthenticationUtil.authenticate(response);
```
Comment (anttsou-google) [inline backend/src/main/java/com/google/step/servlets/WebhookServlet.java:59]: just inline this since the variable name is not being re-used.

**Comment by FluffyPenguin** [comment]
Fixed nits with a few other changes.
Changed UserAuthenticationUtil methods to static
Renamed UserDatastoreUtil to AdvertiserUtil

File: backend/src/main/java/com/google/step/servlets/AuthenticationServlet.java (hunk 1 above)
Comment (FluffyPenguin) [reply backend/src/main/java/com/google/step/servlets/AuthenticationServlet.java]: fixed

File: backend/src/main/java/com/google/step/servlets/WebhookServlet.java (hunk 2 above)
Comment (FluffyPenguin) [reply backend/src/main/java/com/google/step/servlets/WebhookServlet.java]: fixed

File: backend/src/main/java/com/google/step/utils/AdvertiserUtil.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+     * @return     true if the user is contained in the datastore as an advertiser, false otherwise
+     * @throws DatastoreFailureException if a datastore error occurs
+     */
+    public static boolean advertiserExistsInDatastore(User user) {
+        try {
+            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+            Key advertiserKey = createAdvertiserKey(user);
+            datastore.get(advertiserKey);
+            return true;
+        } catch (EntityNotFoundException e) {
```
Comment (anttsou-google) [inline backend/src/main/java/com/google/step/utils/AdvertiserUtil.java:47]: to be safe, add a finally and maybe return false.

Comment (FluffyPenguin) [reply backend/src/main/java/com/google/step/utils/AdvertiserUtil.java:47]: Will do in the next PR!

**Pull request merged**